// squar_modules/index.js
var AvlTree = require("@tyriar/avl-tree");
var roomTree = new AvlTree();
var tableTree = new AvlTree();
var userTree = new AvlTree();
var roomsDetails = [
    {
        "roomId":1,
        "roomName":"1. Oda (100 Kp)",
        "minKp":100,
        "tables":new AvlTree()
    },
    {
        "roomId":2,
        "roomName":"2. Oda (200 Kp)",
        "minKp":200,
        "tables":new AvlTree()
    },
    {
        "roomId":3,
        "roomName":"3. Oda (300 Kp)",
        "minKp":300,
        "tables":new AvlTree()
    },
    {
        "roomId":4,
        "roomName":"4. Oda (400 Kp)",
        "minKp":400,
        "tables":new AvlTree()
    },
    {
        "roomId":5,
        "roomName":"5. Oda (500 Kp)",
        "minKp":500,
        "tables":new AvlTree()
    }
];
var roomNode = function(){
	this.roomId = 0;
	this.roomName = "";
	this.minKp = 0;
	this.tables = "";
}
var tableNode = function(){
	this.tableId = 0;
	this.usersmd5 = ["",""];
	this.game = undefined;
	this.turn = -1;
	this.poolKp = 0;
}
var squarNode = function(){
	this.x = 0;
	this.y = 0;
	this.rightLine = false;
	this.bottomLine = false;
	this.leftLine = false;
	this.topLine = false;
}
var gameNode = function(){
	this.width = 0;
	this.squars = new AvlTree();
	this.create = function(){
		for(var i=0; i< this.width; i++){
			for(var j=0; j< this.width; j++){
				
			}
		}
	}
}
for(var i=0;i<roomsDetails.length;i++){
    var newRoom = new roomNode();
    newRoom.roomId = roomsDetails[i].roomId;
    newRoom.roomName = roomsDetails[i].roomName;
    newRoom.minKp = roomsDetails[i].minKp;
    newRoom.tables = roomsDetails[i].tables;
    roomTree.insert(roomsDetails[i].roomId,newRoom)
}
var table = new tableNode();
module.exports = {
	"rooms":roomTree,
	"users":userTree
};
